bin/tsc

```diff
@@ -1,2 +1,13 @@
 #!/usr/bin/env node
-require('../lib/tsc.js')
+var resolve = require('resolve').sync;
+var path = require('path');
+var basedir = process.cwd();
```

Comment:
Yes, I think that commonly happens when you e.g. have an editor integration, and the editor effectively runs `/usr/bin/clang-format some/path/to/my/file.ts`. I'm not sure how much that applies to TypeScript.

I think -p is a very good signal, but of course at some point you end up having to parse all of `tsc`'s args, which gets nasty. Some sort of a simple heuristic like `-p` or `--project` followed by a file name could work, though. If you want to be correct, you could use TypeScript's `ts.parseCommandLine` API, which should give a precise result.